src/bss.rs

```diff
@@ -1,112 +1,99 @@
         };
 
         if let Some(beacon_interval) = &self.beacon_interval {
-            result.push(format!(
-                "beacon interval : {} TUs",
-                parse_u16(beacon_interval)
-            ))
+            result.push(format!("beacon interval : {} TUs", beacon_interval))
         };
 
         if let Some(seen_ms_ago) = &self.seen_ms_ago {
-            result.push(format!("last seen : {} ms", parse_u32(seen_ms_ago)))
+            result.push(format!("last seen : {} ms", seen_ms_ago))
         };
 
         if let Some(status) = &self.status {
-            result.push(format!("status : {}", parse_u32(status)))
+            result.push(format!("status : {}", status))
         };
 
-        if let Some(signal) = &self.signal {
-            result.push(format!(
-                "signal : {:?} dBm",
-                parse_i32(signal) as f32 / 100.00
-            ))
+        if let Some(signal) = self.signal {
+            result.push(format!("signal : {:?} dBm", signal as f32 / 100.00))
         };
 
         write!(f, "{}", result.join("\n"))
     }
 }
 
-impl ParseNlAttr for Bss {
+impl FromNlAttributeHandle for Bss {
     /// Parse netlink messages returned by the nl80211 command CmdGetScan
-    fn parse(&mut self, handle: AttrHandle<Nl80211Attr>) -> Bss {
+    fn from_handle(handle: AttrHandle<Nl80211Attr>) -> Result<Bss, NlError> {
+        let mut bss = Bss {
+            ..Default::default()
+        };
         for attr in handle.iter() {
             println!("{:?}", attr);
-            match attr.nla_type {
-                Nl80211Attr::AttrBss => {
-                    let sub_handle = attr.get_nested_attributes::<Nl80211Bss>().unwrap();
-                    for sub_attr in sub_handle.iter() {
-                        match sub_attr.nla_type {
-                            Nl80211Bss::BssBeaconInterval => {
-                                self.beacon_interval = Some(sub_attr.payload.clone())
-                            }
-                            Nl80211Bss::BssFrequency => {
-                                self.frequency = Some(sub_attr.payload.clone())
-                            }
-                            Nl80211Bss::BssSeenMsAgo => {
-                                self.seen_ms_ago = Some(sub_attr.payload.clone())
-                            }
-                            Nl80211Bss::BssStatus => self.status = Some(sub_attr.payload.clone()),
-                            Nl80211Bss::BssBssid => self.bssid = Some(sub_attr.payload.clone()),
-                            Nl80211Bss::BssSignalMbm => {
-                                self.signal = Some(sub_attr.payload.clone())
-                            }
-                            _ => (),
-                        }
+
+            if attr.nla_type != Nl80211Attr::AttrBss {
+                continue;
+            }
+
+            let sub_handle = attr.get_nested_attributes::<Nl80211Bss>().unwrap();
```

Comment:
piggy backing on  @mtp401's comment regarding panics:
We should handle the error here rather than unwrapping/panicking